.github/workflows/cd-deploy-nodes-gcp.yml

```
@@ -118,99 +98,37 @@ jobs:
 
   # Test that Zebra works using the default config with the latest Zebra version.
   test-configuration-file:
-    name: Test Zebra CD Docker config file
-    timeout-minutes: 15
-    runs-on: ubuntu-latest
+    name: Test Zebra default Docker config file
     needs: build
-    steps:
-      - uses: r7kamura/rust-problem-matchers@v1.4.0
-
-      - name: Inject slug/short variables
-        uses: rlespinasse/github-slug-action@v4
-        with:
-          short-length: 7
-
-      # Make sure Zebra can sync at least one full checkpoint on mainnet
-      - name: Run tests using the default config
-        shell: /usr/bin/bash -exo pipefail {0}
-        run: |
-          docker pull ${{ vars.GAR_BASE }}/zebrad@${{ needs.build.outputs.image_digest }}
-          docker run --detach --name default-conf-tests -t ${{ vars.GAR_BASE }}/zebrad@${{ needs.build.outputs.image_digest }}
-
-          # Use a subshell to handle the broken pipe error gracefully
-          (
-            trap "" PIPE;
-            docker logs \
-            --tail all \
-            --follow \
-            default-conf-tests | \
-            tee --output-error=exit /dev/stderr | \
-            grep --max-count=1 --extended-regexp --color=always \
-            -e "net.*=.*Main.*estimated progress to chain tip.*BeforeOverwinter"
-          ) || true
-          LOGS_EXIT_STATUS=$?
-
-          docker stop default-conf-tests
-
-          EXIT_STATUS=$(docker wait default-conf-tests || echo "Error retrieving exit status");
-          echo "docker exit status: $EXIT_STATUS";
-
-          # If grep found the pattern, exit with the Docker container exit status
-          if [ $LOGS_EXIT_STATUS -eq 0 ]; then
-              exit $EXIT_STATUS;
-          fi
-
-          # Handle other potential errors here
-          echo "An error occurred while processing the logs.";
-          exit 1;
+    uses: ./.github/workflows/sub-test-zebra-config.yml
+    with:
+      test_id: 'default-conf'
+      grep_patterns: '-e "net.*=.*Main.*estimated progress to chain tip.*BeforeOverwinter"'
+      test_variables: '-e NETWORK'
+      network: 'Mainnet'
 
-  # Test reconfiguring the docker image for testnet.
+  # Test reconfiguring the the docker image for tesnet.
   test-configuration-file-testnet:
-    name: Test testnet Zebra CD Docker config file
-    timeout-minutes: 15
-    runs-on: ubuntu-latest
+    name: Test Zebra testnet Docker config file
```

The production image name needs to be different from the CI (test) image jobs.

```suggestion
    name: Test CD testnet Docker config file
```